lib/ansible/galaxy/role.py

```diff
@@ -379,7 +379,11 @@ def install(self):
                                     if n_part != '..' and not n_part.startswith('~') and '$' not in n_part:
                                         n_final_parts.append(n_part)
                                 member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
+                                # deprecated: description='extract fallback without filter' python_version='3.11'
+                                if hasattr(tarfile, 'data_filter'):
+                                    role_tar_file.extract(member, to_native(self.path), filter='data')
```

Comment:
Even though `mypy` currently doesn't complain about this (since the method isn't being checked), it would be a good idea to add a type ignore here:

```suggestion
                                    role_tar_file.extract(member, to_native(self.path), filter='data')  # type: ignore[call-arg]
```

That way we can enable checking (in a separate PR) by annotating this function with a return type, and it won't introduce any new errors. I've already verified it is reported with the return type annotation present.

Reply:
In case anyone is wondering, the same is *not* needed for the integration test, since our `mypy` sanity test isn't used to check code in integration tests.